packages/modules/common/component_state.py

```diff
@@ -1,36 +1,97 @@
 from typing import List
+from abc import abstractmethod
 
 
-class BatState:
-    def __init__(self, power: float, soc: int, imported: float, exported: float):
-        self.power = power
-        self.soc = soc
-        self.imported = imported
-        self.exported = exported
-
-
-class CounterState:
-    def __init__(self,
-                 voltages: List[float],
-                 currents: List[float],
-                 powers: List[float],
-                 power_factors: List[float],
-                 imported: float,
-                 exported: float,
-                 power_all: float,
-                 frequency: float):
-        self.voltages = voltages
-        self.currents = currents
-        self.powers = powers
-        self.power_factors = power_factors
-        self.imported = imported
-        self.exported = exported
-        self.power_all = power_all
-        self.frequency = frequency
-
-
-class InverterState:
-    def __init__(self, power: float, counter: float, currents: List[float]):
-        self.power = power
-        self.counter = counter
-        self.currents = currents
+class BatUpdater:
+    class State:
+        def __init__(self, imported: float, exported: float, power: float,
+                     soc: int):
+            self.imported = imported
+            self.exported = exported
+            self.power = power
+            self.soc = soc
+
+    def __init__(self, delegate):
+        self.delegate = delegate
+
+    def get_values(self) -> State:
+        return self.delegate.get_values()
+
+
+class AbstractBat:
+    @abstractmethod
+    def __init__(self, device_id: int, component_config: dict, tcp_client,
+                 sim_count) -> None:
+        pass
+
+    @abstractmethod
+    def get_values(self) -> BatUpdater.State:
+        pass
+
+
+class CounterUpdater:
+    class State:
+        def __init__(self,
+                     imported: float,
+                     exported: float,
+                     power_all: float,
+                     voltages: List[float] = [0] * 3,
+                     currents: List[float] = [0] * 3,
+                     powers: List[float] = [0] * 3,
+                     power_factors: List[float] = [0] * 3,
```

Comment:
Ich denke in diesen Fällen wäre doch ein `Tuple[float]` besser als ein `List[float]`:

```
>>> def my_func(x = [1, 2, 3]):
...     for i in range(len(x)):
...         x[i] *= 2
...     print(x)
... 
>>> my_func()
[2, 4, 6]
>>> my_func()
[4, 8, 12]
```
Das Problem ist, dass die Defaultwerte für die Argumente nur einmal von Python initialisiert werden und bei folgenden Aufrufen immer wieder die selbe Referenz reingegeben wird. Manchmal kann das sogar praktisch sein, aber in den meisten Fällen würde ich behaupten führt das eher zu Verwirrung. Es ist zwar "unwahrscheinlich" das jemand die Liste selbst irgendwo verändert, aber wenn doch, dann ist die Verwunderung groß und die Fehlersuche langwierig. Daher besser das Problem von vornherein vermeiden.

Weil Tuples immutable sind, haben die das Problem nicht.

Initialisierung ginge entweder so:
```python
def my_func(voltages: Tuple[float] = (0,)*3)
```
oder so:
```python
def my_func(voltages: Tuple[float] = (0, 0, 0))
```

Wenn du doch bei der `list` bleiben wäre es möglich als Defaultwert `None` zu nehmen und dann im Code unten
```python
if voltages is None:
    voltages = [0, 0, 0]
```